### PR TITLE
refactor: spin

### PR DESCRIPTION
**Pull request by aibayanyu20**
First of all, thank you for your contribution! 😄

New feature please send pull request to feature branch, and rest to master branch. Pull request will be merged after one of collaborators approve. Please makes sure that these form are filled before submitting your pull request, thank you!

[[中文版模板 / Chinese template](https://github.com/vueComponent/ant-design-vue/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [x] Refactoring
- [ ] Code style optimization
- [ ] Branch merge
- [ ] Other (about what?)

### What's the background?

> Refactor Spin for v4

### API Realization (Optional if not new feature)

> 1. Basic thought of solution and other optional proposal.
> 2. List final API realization and usage sample.
> 3. GIF or snapshot should be provided if includes UI/interactive modification.

### What's the effect? (Optional if not new feature)

> 1. Does this PR affect user? Which part will be affected?
> 2. What will say in changelog?
> 3. Does this PR contains potential break change or other risk?

### Changelog description (Optional if not new feature)

> 1. English description
> 2. Chinese description (optional)

### Self Check before Merge

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

### Additional Plan? (Optional if not new feature)

> If this PR related with other PR or following info. You can type here.

### CODE REVIEW

File: components/spin/Spin.tsx [hunk 1]
```diff
@@ -40,133 +41,101 @@ export default defineComponent({
...
-    renderIndicator(prefixCls: string) {
-      const dotClassName = `${prefixCls}-dot`;
-      let indicator = getComponent(this, 'indicator');
-      // should not be render default indicator when indicator value is null
-      if (indicator === null) {
-        return null;
+  setup(props, { attrs, slots }) {
+    const { prefixCls, size, direction } = useConfigInject('spin', props);
+    const [wrapSSR, hashId] = useStyle(prefixCls);
+    const children = slots.default?.();
```
Comment (tangjinzhou) [inline components/spin/Spin.tsx:47]: 放在return 里面，不然插槽内容不更新吧

File: components/spin/Spin.tsx [hunk 2]
```diff
@@ -40,133 +41,101 @@ export default defineComponent({
...
+      }
+      const spinElement = (
+        <div {...divProps} class={spinClassName} aria-live="polite" aria-busy={sSpinning.value}>
+          {renderIndicator(prefixCls.value)}
+          {tip ? <div class={`${prefixCls.value}-text`}>{tip}</div> : null}
         </div>
       );
-    }
-    return spinElement;
+      if (children && children.length) {
```
Comment (tangjinzhou) [inline components/spin/Spin.tsx:124]: 执行下 filterEmpty

**Review by tangjinzhou — CHANGES_REQUESTED** [review]

File: components/spin/Spin.tsx (hunk 1 above)
Comment (aibayanyu20) [reply components/spin/Spin.tsx:47]: OK，这里疏忽了

**Comment by tangjinzhou** [comment]
https://github.com/vueComponent/ant-design-vue/discussions/6212   在这里留言备注下吧，免得被多人重复开发

**Pull request merged**